src/NCalc.Core/Parser/NCalcGrammar.cs

```diff
@@ -0,0 +1,292 @@
+
+        [GenerateParser]
+        public static Parser<string> NotEqualParser() =>
+            OneOf(Terms.Text("<>"), Terms.Text("!="));
+
+        [GenerateParser]
+        public static Parser<string> InPrser() => Terms.Text("in", true);
+
+        [GenerateParser]
+        public static Parser<string> NotInParser() => Terms.Text("not in", true);
+
+        [GenerateParser]
+        public static Parser<string> LikeParser() => Terms.Text("like", true);
+
+        [GenerateParser]
+        public static Parser<string> NotLikeParser() => Terms.Text("not like", true);
+
+        [GenerateParser]
+        public static Parser<string> GreaterParser() => Terms.Text(">");
+
+        [GenerateParser]
+        public static Parser<string> GreaterOrEqualParser() => Terms.Text(">=");
+
+        [GenerateParser]
+        public static Parser<string> LesserParser() => Terms.Text("<");
+
+        [GenerateParser]
+        public static Parser<string> LesserOrEqualParser() => Terms.Text("<=");
+
+        [GenerateParser]
+        public static Parser<string> LeftShiftParser() => Terms.Text("<<");
+
+        [GenerateParser]
+        public static Parser<string> RightShiftParser() => Terms.Text(">>");
+
+        [GenerateParser]
+        public static Parser<string> ExponentParser() => Terms.Text("**");
+
+        [GenerateParser]
+        public static Parser<char> OpenParenParser() => Terms.Char('(');
+
+        [GenerateParser]
+        public static Parser<char> CloseParenParser() => Terms.Char(')');
+
+        [GenerateParser]
+        public static Parser<char> OpenBraceParser() => Terms.Char('[');
+
+        [GenerateParser]
+        public static Parser<char> CloseBraceParser() => Terms.Char(']');
+
+        [GenerateParser]
+        public static Parser<char> OpenCurlyBraceParser() => Terms.Char('{');
+
+        [GenerateParser]
+        public static Parser<char> CloseCurlyBraceParser() => Terms.Char('}');
+
+        [GenerateParser]
+        public static Parser<char> QuestionMarkParser() => Terms.Char('?');
+
+        [GenerateParser]
+        public static Parser<char> ColonParser() => Terms.Char(':');
+
+        [GenerateParser]
+        public static Parser<char> SharpTermsParser() => Terms.Char('#');
+
+        [GenerateParser]
+        public static Parser<char> SharpLiteralsParser() => Literals.Char('#');
+
+        [GenerateParser]
+        public static Parser<TextSpan> IdentifierParser() => Terms.Identifier();
+
+        [GenerateParser]
+        public static Parser<string> NotParser()
+        {
+            var openParen = OpenParenParser();
+
+            return OneOf(
+                Terms.Text("NOT", true)
+                .AndSkip(OneOf(Literals.WhiteSpace(), Not(AnyCharBefore(openParen)))),
+                Terms.Text("!"));
+        }
+
+        [GenerateParser]
+        public static Parser<string> AndParser() =>
+            OneOf(Terms.Text("AND", true), Terms.Text("&&"));
```

Comment:
One thing to check in the generated source is that usages of OneOf/Or are converted to switch/cases. Not just this one. That will validate the decomposition of smaller generated parser is actually supported.

Reply:
I think outer parsers might still generate and contain the same code ... Not a big deal but not sure having a generated version for each node makes sense. Can't be less than one, but less is better.